File: jvm/core/main/scala/com/webank/eggroll/core/deepspeed/job/meta/QueryJobStatusResponse.scala
package com.webank.eggroll.core.deepspeed.job.meta

import com.google.protobuf.ByteString
import com.webank.eggroll.core.meta.Deepspeed

import scala.language.implicitConversions

case class QueryJobStatusResponse(sessionId: String, status: String)

object QueryJobStatusResponse {
  implicit def serialize(src: QueryJobStatusResponse): Array[Byte] = {
    Deepspeed.QueryJobStatusResponse.newBuilder()
      .setSessionId(src.sessionId)
      .setStatus(src.status)
      .build()
      .toByteArray
  }

  implicit def deserialize(byteString: ByteString): QueryJobStatusResponse = {
    val proto = Deepspeed.QueryJobStatusResponse.parseFrom(byteString)
    QueryJobStatusResponse(
      sessionId = proto.getSessionId,
      status = proto.getStatus
    )
  }
}

File: jvm/new_core/src/main/java/com/eggroll/core/pojo/QueryJobStatusResponse.java
package com.eggroll.core.pojo;

import com.webank.eggroll.core.meta.Deepspeed;
import lombok.Data;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

@Data
public class QueryJobStatusResponse implements RpcMessage{
    Logger log = LoggerFactory.getLogger(QueryJobStatusResponse.class);
    private String sessionId;
    private String status;

    @Override
    public byte[] serialize() {
        Deepspeed.QueryJobStatusResponse.Builder builder = Deepspeed.QueryJobStatusResponse.newBuilder();
        builder.setSessionId(this.sessionId)
                .setStatus(this.status);
        return builder.build().toByteArray();
    }

    @Override
    public void deserialize(byte[] data) {
        try {
            Deepspeed.QueryJobStatusResponse queryJobStatusResponse = Deepspeed.QueryJobStatusResponse.parseFrom(data);
            this.sessionId = queryJobStatusResponse.getSessionId();
            this.status = queryJobStatusResponse.getStatus();
        } catch (Exception e) {
            log.error("deserialize error : ", e);
        }
    }
}
